File: Python/webhook/create_webhook.py
from flask import Flask, request
import requests

app = Flask(__name__)

@app.route('/webhook', methods=['POST'])
def webhook():

    data = request.data.decode("utf-8") 
    if data:
        if 'SubscribeURL' in data:
            subscribe_url = data.split('"SubscribeURL": "')[1].split('"')[0]
            response = requests.get(subscribe_url)
            if response.status_code == 200:
                return 'Subscription activated successfully', 200
            else:
                return 'Failed to activate subscription', 400

        else:
            print(data)
            return 'Request received successfully', 200
    else:
        return 'Empty request', 400

if __name__ == '__main__':
    app.run(host= "localhost" ,port=5000 ,debug=True)

File: JavaScript/create_webhook.js
const express = require('express');
const bodyParser = require('body-parser');
const fetch = require('node-fetch');

const app = express();
const PORT = 5000;

app.use(bodyParser.text());

app.post('/webhook', async (req, res) => {
    const data = req.body;
    if (data) {
        if (data.includes('SubscribeURL')) {
            const subscribeUrl = data.split('"SubscribeURL": "')[1].split('"')[0];
            try {
                const response = await fetch(subscribeUrl);
                if (response.ok) {
                    res.status(200).send('Subscription activated successfully');
                } else {
                    res.status(400).send('Failed to activate subscription');
                }
            } catch (error) {
                console.error('Error occurred:', error);
                res.status(500).send('Internal server error');
            }
        } else {
            console.log(data);
            res.status(200).send('Request received successfully');
        }
    } else {
        res.status(400).send('Empty request');
    }
});

app.listen(PORT, () => {
    console.log(`Server is running on port ${PORT}`);
});
